### DISCUSSION TITLE
Package Google test source code into 2 files

### DISCUSSION THREAD

**Discussion by d7d1cd** [google/googletest#3575] [Community Help]
Good afternoon. I saw information on the Internet that the source codes for Google test using a special script can be packed into 2 files: gtest.h and gtest-all.cc. However, I did not find such a script in the repository. Tell me how to make such a package?

**Comment by joshiayush** [comment]
> I saw information on the Internet that the source codes for Google test using a special script can be packed into 2 files: gtest.h and gtest-all.cc.

@d7d1cd Where exactly did you find such information? You don't need a script for that, the source code for `gtest` is already packed into `gtest.h` and `gtest-all.cc` files, these two files include all the respective headers and object files already.

If I'm not wrong and you want the pre-processed files then you just need to run the pre-processor over the file `gtest-all.cc` and you will get the entire source code for `gtest` packed into one single file. 

This is how you do it but before running the following command you need to `cd` into the nested `googletest` directory:
```shell
$ g++ -Iinclude -E src/gtest-all.cc > one_single_packed_file.cc
```

**Reply by d7d1cd** [reply]
Thank you for your help! I've already figured out this issue...
